components/support/images/src/main/java/mozilla/components/support/images/compose/Client.kt

```diff
@@ -0,0 +1,163 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.images.compose
+
+import android.graphics.Bitmap
+import androidx.compose.material.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.graphics.painter.BitmapPainter
+import androidx.compose.ui.graphics.painter.ColorPainter
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import mozilla.components.concept.fetch.Client
+import mozilla.components.concept.fetch.Request
+import mozilla.components.concept.fetch.isSuccess
+import mozilla.components.support.images.DesiredSize
+import mozilla.components.support.images.decoder.AndroidImageDecoder
+import java.io.IOException
+import java.util.concurrent.TimeUnit
+
+private const val DEFAULT_CONNECT_TIMEOUT = 2L // Seconds
+private const val DEFAULT_READ_TIMEOUT = 10L // Seconds
+private const val DEFAULT_MAXIMUM_SCALE_FACTOR = 2.0f
+private const val DEFAULT_MIN_MAX_MULTIPLIER = 3
+
+private val defaultTargetSize = 100.dp
+
+private val decoders = listOf(
+    AndroidImageDecoder()
+)
+
+/**
+ * Returns a [Painter] that will load an image from [url]. Until loading has completed [placeholder]
+ * will be used. If loading or decoding the image failed then [fallback] is used.
+ *
+ * @param url The URL of the image that should get loaded.
+ * @param placeholder A [Painter] that will be used until the image is loaded.
+ * @param fallback A [Painter] that will be used if loading the image failed.
+ * @param private Whether or not this is a private request. Like in private browsing mode, private
+ * requests will not cache anything on disk and not send any cookies shared with the browser.
+ * @param connectTimeout A timeout to be used when connecting to the remote server.  If the timeout
+ * expires before the connection can be established then [fallback] will get used.
+ * @param readTimeout A timeout to be used when reading from the remote server. If the timeout
+ * expires before there is data available for read then [fallback] will get used.
+ * @param targetSize The target image size, the loaded image should be scaled to.
+ * @param minSize The minimum size before an image will be considered too small and [fallback] will
+ * be used instead.
+ * @param maxSize The maximum size before an image will be considered too large and [fallback] will
+ * be used instead.
+ * @param maxScaleFactor The maximum factor a loaded image will be scaled up or down by until
+ * [fallback] will be used.
+ */
+@Composable
+@Suppress("LongParameterList")
```

Comment:
Should we bump the value for this warning? With compose, we're probably going to be suppressing it for every method.